types/draft-js/index.d.ts

```diff
@@ -497,16 +505,13 @@ declare namespace Draft {
              *   - "props": Props to be passed into the React component that will be used
              *     merged with DraftDecoratorComponentProps
              */
-            interface DraftDecorator<P = any> {
+            interface DraftDecorator<P = {}> {
                 strategy: (
                     block: ContentBlock,
                     callback: (start: number, end: number) => void,
                     contentState: ContentState,
                 ) => void;
-                component:
-                    | React.Component
-                    | React.ComponentClass<any>
-                    | ((props: DraftDecoratorComponentProps & P) => React.ReactNode);
+                component: DraftDecoratorComponent;
```

Comment:
Can’t comment on whether the overall change looks good, but this was probably intended to be `DraftDecoratorComponent<P>`